src/components/AccessCodeModal.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useCallback, useState } from 'react';
+
+import { Button, Container, Input, InputProps, Modal, Text } from '@zextras/carbonio-design-system';
+import { useTranslation } from 'react-i18next';
+
+interface AccessCodeModalProps {
+	queryWithAccessCode: (accessCode: string) => void;
+	wrongAccessCode: boolean;
+}
+
+export const AccessCodeModal = ({
+	wrongAccessCode,
+	queryWithAccessCode
+}: AccessCodeModalProps): React.JSX.Element => {
+	const [t] = useTranslation();
+
+	const [accessCode, setAccessCode] = useState('');
+	const [isAccessCodeShown, setIsAccessCodeShown] = useState(false);
+
+	const toggleShowAccessCode = useCallback(() => {
+		setIsAccessCodeShown((prevState) => !prevState);
+	}, []);
+
+	const CustomElement = useCallback<NonNullable<InputProps['CustomIcon']>>(
+		() => (
+			<Button
+				type={'ghost'}
+				color={wrongAccessCode ? 'error' : 'text'}
+				size={'extralarge'}
+				icon={isAccessCodeShown ? 'EyeOutline' : 'EyeOffOutline'}
+				onClick={toggleShowAccessCode}
+			/>
+		),
+		[isAccessCodeShown, toggleShowAccessCode, wrongAccessCode]
+	);
+
+	const inputOnChange = useCallback<NonNullable<InputProps['onChange']>>((e) => {
+		setAccessCode(e.target.value);
+	}, []);
+
+	const onConfirm = useCallback(() => {
+		queryWithAccessCode(accessCode);
+	}, [accessCode, queryWithAccessCode]);
+
+	return (
+		<Modal
+			showCloseIcon={false}
+			title={t(
+				'carbonio-public-folder-ui.accessCode.modal.title',
+				'The link is secured by an access code'
+			)}
+			open
+			confirmLabel={t('carbonio-public-folder-ui.accessCode.modal.button.confirm', 'Done')}
+			onConfirm={onConfirm}
+		>
+			<Container gap={'1rem'} crossAlignment={'flex-start'}>
+				<Text>
+					{t(
+						'carbonio-public-folder-ui.accessCode.modal.description',
+						'Please, insert the access code to view the folder'
+					)}
+				</Text>
+				<Input
+					type={isAccessCodeShown ? 'text' : 'password'}
+					label={t('publicLink.accessCode.input.label', 'Access code')}
+					value={accessCode}
+					onChange={inputOnChange}
```

Comment:
Since we are not doing validation on input, we could use the uncontrolled mode and let the input do its stuff, and then retrieve the value through the ref on submit